### ISSUE TITLE
Failures on [cfe] Handle == on Never

### ISSUE THREAD

**Issue by johnniwinther** [dart-lang/sdk#41986]
There are new test failures on [[cfe] Handle == on Never](https://dart-review.googlesource.com/c/sdk/+/147081).

The tests
```
language/nnbd/never/never_null_assignability_strong_test RuntimeError (expected Pass)
```
are failing on configurations
```
dart2js-hostasserts-strong-linux-x64-d8
```

**Comment by johnniwinther** [comment]
This changes from `Crash` to `RuntimeError` on this CL.

**Comment by fishythefish** [comment]
`RuntimeError` is fine for now - all strong mode failures will eventually be addressed in the burndown.

**Comment by fishythefish** [comment]
Fixed by bd672a90af7d3f456c60e2e30f5945bc18255a95
